In [1]:
import librosa.display

Load Audiodata

- file-sr: 44100 (standard sr for audio-media)
    - default-sr: 22050

In [3]:
audioPath = "../../soundFiles/TroglodytesTroglodytes/WAV/TroglodytesTroglodytes0.wav"
#audioPath = "../soundFiles/Test/test1.m4a"
    
audioData, sr = librosa.load(audioPath, sr=11025, duration=5)
    
print(sr)
print(len(audioData))

11025
55125


STFT (Short-time fourier transform)

In [4]:
#https://librosa.org/doc/latest/generated/librosa.stft.html
stftData = librosa.stft(audioData)
print(stftData.shape)

(1025, 108)


Examining Data

2. stft-Data

2.1 length

The dimension is always [1025, *], where * is:


Sr-Variation

bei sr 1000 und length 1 ist width = 2

bei sr 2000 und length 1 ist width = 4

bei sr 3000 und length 1 ist width = 6

formula: stftWidth = sr/500


Duration-variation

bei sr 1000 und length 0.5 ist width = 1

bei sr 1000 und length 1 ist width = 2

bei sr 1000 und length 1.5 ist width = 3

bei sr 1000 und length 2 ist width = 4

formula: stftWidth = length/500


Gesamt-Variation

formula: stftWidth = (sr * length)/500

In [13]:
print(len(stftData))
print(len(stftData[0]))

1025
22


In [14]:
#test sftf
# audioData1, sr = librosa.load(audioPath, sr=4000, offset=1, duration=2)
# audioData2, sr = librosa.load(audioPath, sr=5000, offset=1, duration=8)
# audioData3, sr = librosa.load(audioPath, sr=3000, offset=1, duration=2)
# audioData4, sr = librosa.load(audioPath, sr=4000, offset=1, duration=5)
# audioData5, sr = librosa.load(audioPath, sr=8000, offset=1, duration=1)
# audioData6, sr = librosa.load(audioPath, sr=4000, offset=1, duration=2)
# audioData7, sr = librosa.load(audioPath, sr=1000, offset=1, duration=2)
# audioData8, sr = librosa.load(audioPath, sr=7000, offset=1, duration=9)
# audioData9, sr = librosa.load(audioPath, sr=4000, offset=1, duration=2)
# audioData10, sr = librosa.load(audioPath, sr=4000, offset=1, duration=2)

# allaudioData1 = [audioData1, audioData2, audioData3, audioData4, audioData5, audioData6, audioData7, audioData8, audioData9, audioData10]

# for i in allaudioData1:
#     stftDataI = librosa.stft(i)
#     print(len(stftDataI[0]))

#conclusion
# stftLength = (sr * duration)/500
# if sr * duration surpasses a certain size, the dimension is close to the formula-value

2.2 data

The data in audioSamples is a single list. Each element is:

- two complex numbers made of two float numbers
    - these are short-term fourier transform coefficients

In [20]:
#short
print(stftData[20])

[-0.28058645-1.7238815e-17j -0.25969002-1.3875529e-01j
 -0.3924419 -5.2738345e-01j -0.42022553-4.4900650e-01j
 -0.05179632-2.0644993e-01j  0.42615023-2.4174424e-01j
  0.5547653 -3.0082491e-01j  0.25291646-2.4842195e-01j
  0.07060383+9.9487249e-03j  0.17444766+3.8694528e-01j
  0.28262377+4.5384896e-01j  0.19320357+5.0744605e-01j
  0.01675273+6.3334227e-01j -0.17226392+5.4172224e-01j
 -0.44678673+4.5886266e-01j -0.4609587 +3.9555109e-01j
 -0.33702123+2.2375599e-01j -0.28528064-3.7011255e-02j
 -0.24738014-2.8402779e-01j -0.35861164-5.2033043e-01j
 -0.2159954 -5.4606467e-01j  0.17208321-4.7477537e-01j]


In [16]:
#short abs
print(abs(stftData))

[[2.8951474e-02 1.1372781e-02 7.8847678e-03 ... 2.1175219e-02
  1.1358724e-02 7.0599981e-02]
 [3.4855746e-02 1.8518807e-02 3.6574847e-03 ... 2.2360850e-02
  1.5312156e-02 7.6589637e-02]
 [4.1574743e-02 2.5218043e-02 1.1194055e-02 ... 2.0402610e-02
  3.0440327e-03 8.3336465e-02]
 ...
 [1.4047805e-03 6.4218690e-04 1.5672776e-04 ... 3.4335087e-04
  9.7633449e-05 6.2962907e-04]
 [1.4637643e-03 7.8300817e-04 9.7489130e-05 ... 2.6367363e-04
  2.6371109e-04 6.1845378e-04]
 [1.5057886e-03 7.2239299e-04 4.7171093e-06 ... 6.6318076e-05
  1.4198729e-04 4.8929238e-04]]


In [17]:
#long
for i in range(len(stftData)):
    print(stftData[i])

[ 0.02895147+0.j  0.01137278+0.j -0.00788477+0.j -0.00319141+0.j
  0.00133028+0.j -0.00737133+0.j -0.00985981+0.j -0.00322626+0.j
  0.00305312+0.j  0.00328455+0.j -0.00094461+0.j  0.00818517+0.j
  0.0238363 +0.j  0.0119289 +0.j -0.01092637+0.j -0.01117432+0.j
  0.00581168+0.j  0.00152743+0.j -0.02486577+0.j -0.02117522+0.j
 -0.01135872+0.j -0.07059998+0.j]
[-0.03485575+5.4691223e-17j  0.00639131-1.7380949e-02j
  0.00319926-1.7725505e-03j  0.00319496+6.4420672e-03j
 -0.00436514-1.7785156e-03j  0.00361145-7.1561043e-03j
  0.00814579+4.9624071e-03j -0.00042146+1.0059585e-03j
 -0.00148046+5.7599479e-03j -0.0048958 -5.7946797e-03j
  0.00566992+1.1641746e-03j -0.00284094+7.9595400e-03j
 -0.01721569+7.5943656e-03j -0.00944323-1.8969459e-02j
  0.01035756-3.6933764e-03j  0.01173861+3.9189602e-03j
 -0.00967517+1.2894897e-02j -0.00439986-1.6461536e-02j
  0.01718012-1.1032877e-02j  0.01648451+1.5108564e-02j
 -0.01528426-9.2378870e-04j  0.04948977-5.8452852e-02j]
[ 0.04157474+1.1438333e-17j -0.0252